PSReadLine/Completion.cs

```diff
@@ -372,9 +372,15 @@ private void DoReplacementForCompletion(CompletionResult completionResult, Comma
         {
             var replacementText = completionResult.CompletionText;
             int cursorAdjustment = 0;
+            CompletionResultType[] selfContainedCompletionResultTypes = {CompletionResultType.Command,
+                CompletionResultType.Keyword, CompletionResultType.ParameterName, CompletionResultType.ProviderItem};
```

Comment:
Many people do not use a space after some keywords, e.g. `if(`.

Similarly, one may not want a space after a filename, e.g. if I'm copying a file but adding an extension, e.g. `.bak`.

And infrequently one might want to type `:` after a parameter name.